src/construct.ts

```diff
@@ -279,10 +279,8 @@ export class Node {
   }
 
   /**
-   * Adds a metadata entry to this construct.
-   * Entries are arbitrary values and will also include a stack trace to allow tracing back to
-   * the code location for when the entry was added. It can be used, for example, to include source
-   * mapping in CloudFormation templates to improve diagnostics.
+   * Adds a metadata entry to the CDK construct tree. View modification in Cloud Assembly manifest.json.
```

Comment:
This a sequence of true statements, but it doesn't flow well. Try something more like

```
Adds a metadata entry to this construct...

Note that construct metadata is not the same as CloudFormation resource metadata and is never written to the CloudFormation template. 
```

Then mention the manifest.json, but don't just say it was written to the manifest file. Mention that it is conditionally written to the manifest and the name of the prop that controls that.